util.h

```diff
@@ -0,0 +1,56 @@
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include <assert.h>
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define BOARD_SIZE 3
+#define GOAL 3
+#define ALLOW_EXCEED 1
+#define N_GRIDS (BOARD_SIZE * BOARD_SIZE)
+#define GET_INDEX(i, j) ((i) * (BOARD_SIZE) + (j))
+#define GET_COL(x) ((x) % BOARD_SIZE)
+#define GET_ROW(x) ((x) / BOARD_SIZE)
+#define LOOKUP(table, i, j, else_value)                         \
```

Comment:
The macro `LOOKUP` is supposed to appear in C implementation file instead of headers.